packages/next/src/build/webpack/loaders/next-flight-loader/index.ts

```diff
@@ -144,7 +147,28 @@ ${JSON.stringify(ref)},
         }
       }
 
-      return this.callback(null, esmSource, sourceMap)
+      const compilation = this._compilation!
+      const originalSourceURL = ModuleFilenameHelpers.createFilename(
+        module,
+        {
+          moduleFilenameTemplate:
+            'webpack://[namespace]/[resource-path]/__nextjs-internal-proxy.mjs',
+          namespace: '_N_E',
+        },
+        {
+          requestShortener: compilation.requestShortener,
+          chunkGraph: compilation.chunkGraph,
+          hashFunction: compilation.outputOptions.hashFunction,
+        }
+      )
+
+      return this.callback(null, esmSource, {
+        version: 3,
+        sources: [originalSourceURL],
+        // minimal, parseable mappings
```

Comment:
maybe explain why we can get away with emitting no mappings? AFAIU the reasoning is sth like this
```suggestion
        // proxy code is ignore-listed, so we don't need an actual sourcemap.
        // this one is effectively empty, but still parseable.
```

Reply:
(also personally i wouldn't mind moving this to a helper like `createEmptySourceMap(sourceUrl, sourceContent)` to have this explanation in one place, but i'm not gonna die on this hill)

Reply:
It's not an empty sourcemap. It's just the minimal amount of mappings you need for it to be parseable by the sourcemap parsers we encounter. I need to think more about a sensible explainer to justify the abstraction. Until then I rather have it inline to avoid spreading an abstraction that may be incorrect.